src/compas/geometry/curves/nurbs.py

```diff
@@ -0,0 +1,495 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+from math import sqrt
+
+from compas.plugins import pluggable
+from compas.geometry import Point
+from compas.geometry import Frame
+from compas.utilities import linspace
+
+from .curve import Curve
+
+
+@pluggable(category='factories')
+def new_nurbscurve(cls, *args, **kwargs):
+    raise NotImplementedError
```

Comment:
I will keep bugging about this: a specific exception that clearly states this is an extension-point that is filled in by code somewhere else would be very good. A lot of our users read the source code, and this leads them to think the feature is literally not implemented, but that is not really this case for it.

Reply:
will do this in separate pr for all pluggables